### PR TITLE
Download with https proxy if https_proxy is set in environment variable

### PR DESCRIPTION
**Pull request by Changqing-JING**
Current vscode-dts doesn't work when npm install behind proxy. Add a feature to read https_proxy from environment variable and download via the proxy.

### CODE REVIEW

**Comment by Changqing-JING** [comment]
@microsoft-github-policy-service agree

**Comment by sean-mcmanus** [comment]
This fixes the issue at https://github.com/microsoft/vscode-cpptools/issues/10631

**Comment by sean-mcmanus** [comment]
@connor4312 @jrieken Are you two still maintaining this repo? Do you have an ETA on when this can get reviewed?

**Comment by sean-mcmanus** [comment]
@Changqing-JING It seems like the version in package.json may need to be updated to 0.3.4.

**Comment by connor4312** [comment]
I would rather use the battle-tested [`https-proxy-agent`](https://www.npmjs.com/package/https-proxy-agent) module than rolling our own implementation here.

**Comment by Changqing-JING** [comment]
force pushed to use [https-proxy-agent](https://www.npmjs.com/package/https-proxy-agent)

File: index.ts [hunk 1]
```diff
@@ -7,6 +7,8 @@ import os from 'os'
 import prompts from 'prompts'
 import minimist from 'minimist'
 import rimraf from 'rimraf'
+import HttpsProxyAgent from 'https-proxy-agent/dist/agent'
```
Comment (connor4312) [inline index.ts:10]:
```suggestion
import HttpsProxyAgent from 'https-proxy-agent'
```

File: tsconfig.json [hunk 2]
```diff
@@ -1,6 +1,7 @@
 {
   "compilerOptions": {
-    "esModuleInterop": true
+    "esModuleInterop": true,
+    "sourceMap": true
```
Comment (connor4312) [inline tsconfig.json:4]: Please also update the npmignore file to avoid publishing the new sourcemap

Comment (Changqing-JING) [reply tsconfig.json:4]: Done

File: index.ts (hunk 1 above)
Comment (Changqing-JING) [reply index.ts]: Done

**Review by connor4312 — APPROVED** [review]

**Review by rebornix — APPROVED** [review]

**Comment by connor4312** [comment]
Thanks for the PR!

**Comment by Changqing-JING** [comment]
Thank you very much! When is the next release on npm?

**Comment by Changqing-JING** [comment]
@connor4312 Could you help to update v0.3.4 to npm, so that we can use the new version in vscode?

**Pull request merged**